### ISSUE TITLE
Multicall with no specific implementation

### ISSUE THREAD

**Issue by pikonha** [blockful-io/external-resolver#148]
### Describe the Refactor Request

The current `multicall` implementation required an specific handling on the client side:

```solidity
function multicall(bytes[] calldata datas)
    external
    returns (bytes[] memory /* results */ )
{
    IWriteDeferral.parameter[] memory params =
        new IWriteDeferral.parameter[](1);

    params[0].name = "data";
    params[0].value = TypeToString.bytesToString(msg.data);

    _offChainStorage(params);
}

```

client handling:

```ts
  let callData
  if (multicall) {
    callData = message.parameters[0].value as `0x${string}`
  } else {
    callData = encodeFunctionData({
      abi: dbABI,
      functionName: message.functionSelector,
      args: message.parameters.map((arg) => arg.value),
    })
  }
```

This happens because of a limitation of the EIP-5559 due to an intermediary structure called `parameters` that has `key` and `value` both as strings to store function arguments.

### Describe Preferred Solution

Find a way to handle the `multicall` the same way as the other calls so the client doesn't need to implement specific handling.

### Describe Alternatives

[change EIP-5559](https://ethereum-magicians.org/t/eip-5559-cross-chain-write-deferral-protocol/10576/13?u=pikonha)

**Comment by pikonha** [comment]
blocked til the changes of the EIP-5559 are accepted by the community

**Comment by pikonha** [comment]
closed by #179
